homeassistant/components/samsungtv/__init__.py

```diff
@@ -62,8 +64,33 @@ async def async_setup(hass, config):
 
 async def async_setup_entry(hass, entry):
     """Set up the Samsung TV platform."""
+    undo_listener = entry.add_update_listener(_update_listener)
```

Comment:
Do we need to unload and setup the entry again? The action is created at runtime and saved at `hass.data[DOMAIN][ip_address][CONF_ON_ACTION]`. Why do we not override it and use this directly in the entity?

Reply:
Interesting point. TBH I haven't thought of this direction until you wrote this.
Updating `hass.data[DOMAIN][ip_address][CONF_ON_ACTION]` won't be enough - the entity currently gets its processed `on_script` on initialization. We can have the entity listen to the update and handle accordingly. It might be a bit simpler, but potentially a bit more error prone.
The advantages of the unload/reload approach are:
1. Works through standard HA mechanisms.
2. Supports other unload scenarios. For example, removing a Samsung TV config entry currently suggests restarting HA. This wont' be needed if the integration can unload. Another example: if in the future we support editing the IP address (see https://github.com/home-assistant/architecture/issues/377), it will work out of the box.

Anyway, pros and cons to both approaches - I'm fine with whatever you choose.

Reply:
Are we going to wait for the linked issue? I would decide for the most future proof way.

Reply:
I don't see a reason to wait - this PR provides benefits as it is, even without the arch issue ever being implemented. The link is just to support the unload route as the more future proof one.